src/main/java/org/apache/ibatis/builder/xml/XMLIncludeTransformer.java

```diff
@@ -70,4 +112,58 @@ private Node findSqlFragment(String refid) {
   private String getStringAttribute(Node node, String name) {
     return node.getAttributes().getNamedItem(name).getNodeValue();
   }
+
+  /**
+   * Add inherited context into newly created one.
+   * @param newContext variables defined current include clause where inherited values will be placed
+   * @param inheritedContext all inherited variables values
+   */
+  private void applyInheritedContext(Properties newContext, Properties inheritedContext) {
+    for (Map.Entry<Object, Object> e : inheritedContext.entrySet()) {
+      if (!newContext.containsKey(e.getKey())) {
+        newContext.put(e.getKey(), e.getValue());
+      }
+    }
+  }
+
+  /**
+   * Read placholders and their values from include node definition. 
+   * @param node Include node instance
+   * @return variables context from include instance (no inherited values)
+   */
+  private Properties getVariablesContext(Node node) {
+    List<Node> subElements = getSubElements(node);
```

Comment:
Redundant? Do you mean to inline the code into `getVariablesContext()` without having separate method used only once?